src/main/ruby/core/io.rb

```diff
@@ -140,7 +141,12 @@ def exhausted?
     # the IO, and +unshift+ again beginning at +start_pos+.
     def unshift(str, start_pos)
       @write_synced = false
-      written = Truffle.invoke_primitive :iobuffer_unshift, self, str, start_pos, @used
+      available = SIZE - @used
```

Comment:
If you wouldn't mind, please update the description for this method. The existing docs are not very well written and a bit misleading. At first I read them as a simple calculation of how many bytes could be written, not as an operation with side effects.

Reply:
I don't really understand this code, I just translated from Java. A InternalBuffer cleanup would be great, but I believe out of scope of this PR.